globus_sdk/base.py

```diff
@@ -55,7 +58,9 @@ class BaseClient(object):
     BASE_USER_AGENT = 'globus-sdk-py-{0}'.format(__version__)
 
     def __init__(self, service, environment=None, base_url=None,
-                 base_path=None, authorizer=None, app_name=None):
+                 base_path=None, authorizer=None, app_name=None,
+                 http_timeout=None,
```

Comment:
I think our usage of `None` is going to be problematic in that we've lost the ability to tell Requests to not timeout. Requests uses `timeout=None` but we'll convert `None` to `60`.

Reply:
Is that... actually a problem? I mean, will anyone actually want to set no-timeout?

I see a couple of okay workarounds:
 - We could translate `-1` to no timeout
 - We could make `http_timeout` part of `**kwargs`, and treat its absence and its presence (with a value of `None`) differently

Do you like either of these? (I have no strong feelings)

Reply:
It _shouldn't_ be a problem but the world is a crazy place and someone, maybe even someone we know, might end up with a service that takes more than 60 seconds to respond...

`-1` for no timeout seems intuitive and is easy to document. I'd probably go with that.